pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go

```diff
@@ -57,44 +68,59 @@ type Endpoint struct {
 	// +kubebuilder:validation:MinLength=1
 	Address string `json:"address"`
 	// Port of the endpoint
-	Port EndpointPort `json:"port"`
+	Port *EndpointPort `json:"port,omitempty"`
```

Comment:
Should it be an `EndpointPort` if it's in the match? Maybe just `Port` is enough